exp/api/v1beta2/rosamachinepool_types.go

```diff
@@ -45,37 +46,64 @@ type RosaMachinePoolSpec struct {
 	// +optional
 	AvailabilityZone string `json:"availabilityZone,omitempty"`
 
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="subnet is immutable"
+	//
+	// +immutable
 	// +optional
 	Subnet string `json:"subnet,omitempty"`
 
 	// Labels specifies labels for the Kubernetes node objects
 	// +optional
 	Labels map[string]string `json:"labels,omitempty"`
 
+	// Taints specifies the taints to apply to the nodes of the machine pool
+	// +optional
+	Taints []RosaTaint `json:"taints,omitempty"`
+
 	// AutoRepair specifies whether health checks should be enabled for machines
 	// in the NodePool. The default is false.
 	// +optional
 	// +kubebuilder:default=false
 	AutoRepair bool `json:"autoRepair,omitempty"`
 
+	// +kubebuilder:validation:Required
+	//
 	// InstanceType specifies the AWS instance type
-	InstanceType string `json:"instanceType,omitempty"`
+	InstanceType string `json:"instanceType"`
 
 	// Autoscaling specifies auto scaling behaviour for this MachinePool.
 	// required if Replicas is not configured
 	// +optional
 	Autoscaling *RosaMachinePoolAutoScaling `json:"autoscaling,omitempty"`
 
-	// TODO(alberto): Enable and propagate this API input.
-	// Taints           []*Taint                     `json:"taints,omitempty"`
-	// TuningConfigs    []string                     `json:"tuningConfigs,omitempty"`
-	// Version          *Version                     `json:"version,omitempty"`
+	// TuningConfigs specifies the names of the tuning configs to be applied to this MachinePool.
+	// Tuning configs must already exist.
+	// +optional
+	TuningConfigs []string `json:"tuningConfigs,omitempty"`
 
 	// ProviderIDList contain a ProviderID for each machine instance that's currently managed by this machine pool.
 	// +optional
 	ProviderIDList []string `json:"providerIDList,omitempty"`
 }
 
+type RosaTaint struct {
+	// +kubebuilder:validation:Required
+	//
+	// The taint key to be applied to a node.
+	Key string `json:"key"`
```

Comment:
We usually have the `+kubebuilder` tags after the comment, like:

```suggestion
	// The taint key to be applied to a node.
	//
	// +kubebuilder:validation:Required
	Key string `json:"key"`
```